### PR TITLE
opam 2.2.0-alpha0-20221104

### PR DESCRIPTION
**Pull request by jonahbeckford**
- [Y] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [Y] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [Y] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [Y] Have you tested your manifest locally with `winget install --manifest <path>`?
- [Y] Does your manifest conform to the [1.2 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.2.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/88782)

### CODE REVIEW

File: manifests/d/Diskuv/opam/2.2.0-alpha0-20221104/Diskuv.opam.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  - interactive
+InstallerSwitches:
+  Silent: --ci --quiet --color=never
+  SilentWithProgress: --ci
+  Interactive: --color=always
+  InstallLocation: --prefix "<INSTALLPATH>"
+AppsAndFeaturesEntries:
+  # Keep in sync with installer\src\private_common.ml which is used by
+  # https://github.com/diskuv/dkml-install-api/blob/main/package/console/common/Windows_registry.ml
+  - Publisher: Diskuv, Inc.
```
Comment (russellbanks) [inline manifests/d/Diskuv/opam/2.2.0-alpha0-20221104/Diskuv.opam.installer.yaml:32]:
```suggestion
```

This matches `Publisher: Diskuv, Inc.` in the locale manifest, so it isn't needed here.

File: manifests/d/Diskuv/opam/2.2.0-alpha0-20221104/Diskuv.opam.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    DisplayName: Diskuv OCaml
```
Comment (russellbanks) [inline manifests/d/Diskuv/opam/2.2.0-alpha0-20221104/Diskuv.opam.installer.yaml:33]:
```suggestion
```

This has to match control panel:

WindowsSandboxClient_xyUnQLVzk4

The `PackageName: opam` in the locale manifest is correct, so this DisplayName is not needed here.

File: manifests/d/Diskuv/opam/2.2.0-alpha0-20221104/Diskuv.opam.installer.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    DisplayVersion: 2.2.0~alpha0~20221104 # BUMP
```
Comment (russellbanks) [inline manifests/d/Diskuv/opam/2.2.0-alpha0-20221104/Diskuv.opam.installer.yaml:34]:
```suggestion
```

The version of the package matches Control Panel, so this DisplayVersion is incorrect and is not needed.

File: manifests/d/Diskuv/opam/2.2.0-alpha0-20221104/Diskuv.opam.installer.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+InstallerType: exe
+InstallModes:
+  - silent
+  - interactive
+InstallerSwitches:
+  Silent: --ci --quiet --color=never
+  SilentWithProgress: --ci
+  Interactive: --color=always
+  InstallLocation: --prefix "<INSTALLPATH>"
+AppsAndFeaturesEntries:
```
Comment (russellbanks) [inline manifests/d/Diskuv/opam/2.2.0-alpha0-20221104/Diskuv.opam.installer.yaml:29]:
```suggestion
```
Not needed if all other AppsAndFeaturesEntries are removed.

File: manifests/d/Diskuv/opam/2.2.0-alpha0-20221104/Diskuv.opam.locale.en-US.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+PackageVersion: 2.2.0-alpha0-20221104 # BUMP
+PackageLocale: en-US
+Publisher: Diskuv, Inc.
+PublisherUrl: https://www.diskuv.com/
+PrivacyUrl: https://diskuv.com/legal#privacy_policy
+PackageName: opam
+PackageUrl: https://opam.ocaml.org/
+License: LGPL-2.1-only WITH OCaml-LGPL-linking-exception
+LicenseUrl: https://github.com/ocaml/opam/blob/master/LICENSE
+Copyright: |
```
Comment (russellbanks) [inline manifests/d/Diskuv/opam/2.2.0-alpha0-20221104/Diskuv.opam.locale.en-US.yaml:14]: The copyright should only really be one line.

**Review by russellbanks — CHANGES_REQUESTED** [review]

**Comment by russellbanks** [comment]
To clarify my requested changes, WinGet matches a package based on what is written to the registry, so the values need to match what is shown in the control panel.

`PackageName` -> DisplayName
`Publisher` -> Publisher
`PackageVersion` -> DisplayVersion

WindowsSandboxClient_xyUnQLVzk4

These are the values that are shown in control panel, so those values in the manifest need to match that. The `AppsAndFeaturesEntries` is usually for if the values that are written to registry do not match what we are using in the manifest.

For example, lots of programs like to put the version in the DisplayName, so we use the default name (for example, `Program-X`) as the package name so that it is consistent across versions, and the actual DisplayName (for example, `Program-X (v1.3.2)`) in the AppsAndFeaturesEntries.

**Comment by jonahbeckford** [comment]
> To clarify my requested changes, WinGet matches a package based on what is written to the registry, ... For example, lots of programs like to put the version in the DisplayName, so we use the default name (for example, `Program-X`) as the package name so that it is consistent across versions, and the actual DisplayName (for example, `Program-X (v1.3.2)`) in the AppsAndFeaturesEntries.

Thanks. Very good to know.

File: manifests/d/Diskuv/opam/2.2.0-alpha0-20221104/Diskuv.opam.locale.en-US.yaml (hunk 5 above)
Comment (jonahbeckford) [reply manifests/d/Diskuv/opam/2.2.0-alpha0-20221104/Diskuv.opam.locale.en-US.yaml]: Changed it. Thanks!

**Review by russellbanks — APPROVED** [review]

**Comment by jonahbeckford** [comment]
One question. I think but do not know for sure that winget follows `semver` ordering. I mean with semver `2.2.0-alpha0-20221104` is less than `2.2.0` ... which would imply that `PackageVersion: 2.2.0` should appear to winget to be an upgrade of `PackageVersion: 2.2.0-alpha0-20221104`. Is that understanding correct?

**Comment by russellbanks** [comment]
@jonahbeckford the documentation for version mapping is [here](https://github.com/microsoft/winget-cli/blob/a989c11e91b78c070bcc97c6ac2c0308a97fb867/doc/specs/%23980%20-%20Apps%20and%20Features%20entries%20version%20mapping.md#solution-design).

From the docs:
> Version 0.0.1-alpha is less than version 0.0.1

So yes, WinGet should treat `2.2.0` as an upgrade for `2.2.0-alpha0-20221104`

**Pull request merged**